src/application/engine/entities/BoDCharacter.java

```diff
@@ -37,6 +37,11 @@
      * @param speed
      *            Movement speed of the character.
      */
+    public BoDCharacter(int id)
```

Comment:
it doesnt matter if its abstract or not or what specialization stats there are present. every character needs the above. so you can define them in the subclasses if that is what you want, but the ctor should require those so that we do not make mistakes in the future